### DISCUSSION TITLE
How to log all models,not just the best/final one, created during tune_model() function?

### DISCUSSION THREAD

**Discussion by raphaelTrench** [pycaret/pycaret#2824] [Q&A]
Hello, I would like to know if it is possible to also register/log every single model that is  tested when executing a hyperparameter search operation with the function "tune_model()"?
If this is not possible today, are there any plans to implement this feature?

**Comment by celestinoxp** [comment]
@ngupta23 Can you answer?

**Comment by Panchiscos** [comment]
Hello there, you can store model quantity in compare_models() and tune_model() in a quite similar way. I'm using this in PyCaret 3.0.4.

Example:

top16 = compare_models(sort='F1', n_select=16)
tuned_top16 = [tune_model(i, optimize='F1') for i in top16]
ranked_tuned_top16 = compare_models(tuned_top16, sort='F1', n_select=16)
evaluate_model(ranked_tuned_top16[0])

Hope this helps :)
